lib/formats/bs_format_ripejson.c

```diff
@@ -0,0 +1,669 @@
+    RDATA->status_msg_state = -1;
+  }
+
+  process_common_fields(format, record);
+
+  return 0;
+}
+
+int process_open_message(bgpstream_format_t *format, bgpstream_record_t *record){
+
+  int json_str_len = STATE->json_fields.body.len;
+  uint8_t* json_str_ptr = STATE->json_fields.body.ptr;
+  int missing_type = 0;
+  int err;
+  uint8_t* ptr = STATE->json_bytes_buffer;
+  int loc = 2;
+  int msg_len = 0;
+
+
+  ptr[loc] = PARSEBGP_BGP_TYPE_OPEN;
+  loc += 1;
+
+  /* add missing open message headers */
+
+  // version
+  ptr[loc] = 4;
+  loc += 1;
+
+  // my autonomous system
+  uint32_t asn4 = strtol((char*)STATE->json_fields.asn.ptr, NULL, 10);
+  uint16_t asn = htons( (uint16_t) (asn4 > UINT16_MAX ?  23456: asn4) );
+  memcpy(&ptr[loc], &asn, sizeof(uint16_t));
+  loc += 2;
+
+  // hold time
+  uint16_t hold_time = htons( (uint16_t) strtol((char*)STATE->json_fields.hold_time.ptr, NULL, 10) );
+  memcpy(&ptr[loc], &hold_time, sizeof(uint16_t));
+  loc += 2;
+
+  // bgp identifier
+  bgpstream_addr_storage_t addr;
+  memcpy(STATE->field_buffer, STATE->json_fields.router_id.ptr, STATE->json_fields.router_id.len);
+  STATE->field_buffer[STATE->json_fields.router_id.len] = '\0';
+  void* rc = bgpstream_str2addr((char*)STATE->field_buffer, &addr);
+  if(rc==NULL){
+    uint32_t router_id = htons(strtoul((char*)STATE->json_fields.router_id.ptr, NULL, 10) );
+    memcpy(&ptr[loc], &router_id, sizeof(uint32_t));
+  } else {
+    memcpy(&ptr[loc], &addr.ipv4, sizeof(uint32_t));
+  }
+  loc += 4;
+
+  // opt parm len
+  if(json_str_ptr[0] == '0' && json_str_ptr[1] == '2'){
+    // if body starts with correct parameter type "0x02"
+    // no missing param type
+    missing_type = 0;
+    msg_len = json_str_len/2;
+    ptr[loc] = (uint8_t) ( msg_len ); // adding
+    loc += 1;
+  } else {
+    // if misses param type
+    missing_type = 1;
+    msg_len = json_str_len/2 + 1 ;
+    ptr[loc] = (uint8_t) ( msg_len ); // adding
+    ptr[loc+1] = (uint8_t) 2;
+    loc += 2;
+  }
+
+  hexstr_to_bytes(&ptr[loc], (char*)STATE->json_fields.body.ptr , STATE->json_fields.body.len);
+
+  // set msg length
+  uint16_t total_length = htons(msg_len+10+2+1);
+  memcpy(&ptr[0], &total_length, sizeof(total_length));
+
+  size_t dec_len = (size_t) total_length;
+
+  RDATA->msg->type = PARSEBGP_MSG_TYPE_BGP;
+  if ((err = parsebgp_decode(STATE->opts, RDATA->msg->type, RDATA->msg,
+                             STATE->json_bytes_buffer, &dec_len)) != PARSEBGP_OK) {
+    fprintf(stderr, "ERROR: Failed to parse message (%d:%s)\n", err, parsebgp_strerror(err));
+    parsebgp_clear_msg(RDATA->msg);
+  }
+
+  // process direction
+  if( bcmp(&"sent", STATE->json_fields.direction.ptr, STATE->json_fields.direction.len) == 0){
+    // equals
+    RDATA->open_msg_direction = 0;
+  } else {
+    RDATA->open_msg_direction = 1;
+  }
+
+  process_common_fields(format, record);
+
+  return 0;
+}
+
+int process_notify_message(bgpstream_format_t *format, bgpstream_record_t *record){
```

Comment:
i think this is unused (which the compiler would have told you about if it was marked static).
since you want it there for future implementation, i'd suggest calling the unsupported message function from this function to make it clear that it is not yet implemented.